Tests/SwiftFormatPrettyPrintTests/GarbageTextTests.swift

```diff
@@ -0,0 +1,147 @@
+fileprivate let bom: Unicode.Scalar = "\u{feff}"
+fileprivate let unknownScalar: Unicode.Scalar = "\u{fffe}"
+
+final class GarbageTextTests: PrettyPrintTestCase {
+  func testHashBang() {
+    let input =
+      """
+      #!/usr/bin/swift -foo -bar
+      print("Hello world!")
+      """
+
+    let expected =
+      """
+      #!/usr/bin/swift -foo -bar
+      print(
+        "Hello world!")
+
+      """
+
+    assertPrettyPrintEqual(input: input, expected: expected, linelength: 20)
+  }
+
+  func testBOM() {
+    let input =
+      """
+      \(bom)print("Hello world!")
+      """
+
+    let expected =
+      """
+      \(bom)print("Hello world!")
+
+      """
+
+    assertPrettyPrintEqual(input: input, expected: expected, linelength: 21)
+  }
+
+  func testBOMPresenceDoesNotPermitLeadingNewlines() {
+    let input =
+      """
+      \(bom)
+      print("Hello world!")
+      """
+
+    let expected =
+      """
+      \(bom)print("Hello world!")
+
+      """
+
+    assertPrettyPrintEqual(input: input, expected: expected, linelength: 21)
+  }
+
+  func testUnknownCodePointAsLeadingTrivia() {
+    let input =
+      """
+      \(unknownScalar)print("Hello world!")
+      """
+
+    let expected =
+      """
+      \(unknownScalar)print("Hello world!")
+
+      """
+
+    assertPrettyPrintEqual(input: input, expected: expected, linelength: 21)
+  }
+
+  func testUnknownCodePointAsTrailingTriviaAreGluedToPreviousToken() {
+    // Note: The third line here is a bit of a weird case. Despite the operator being separated from
+    // the left-hand-side by a space, the intervening garbage text appears to put the parser in a
+    // mode where it treats the operator as a postfix operator, which then causes `secondTerm` to be
+    // parsed as its own distinct statement and `CodeBlockItem`---hence the lack of indentation.
```

Comment:
Wow, what a weird case. Is that worth a bug for the parser?  I'm not sure if it can be fixed, but I would expect any text identified as `garbageText` to not impact parsing.

Reply:
Yeah, I don't know if garbage text affects the trivia in a way that causes it to ignore _all_ of the other nearby trivia for the purposes of associating the operator with the thing before it, or what's going on 🤷‍♂